### PR TITLE
Set up testing environment api

### PR DESCRIPTION
**Pull request by johanperjulius1**
## This PR contains:

- Rspec configuration
- Shoulda Matchers configuration
- Factory Bot configuration
- Configuring RespnseJSON
- Installing Rails Pry
- Cleaning up files

### CODE REVIEW

**Review by SebastianN97 — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
 
+gem "rails", "~> 6.0.3", ">= 6.0.3.4"
+gem "pg", ">= 0.18", "< 2.0"
+gem "puma", "~> 4.1"
+gem "bootsnap", ">= 1.4.2", require: false
+gem "rack-cors", require: "rack/cors"
 
 group :development, :test do
-  gem 'byebug'
+  gem "byebug"
```
Comment (tochman) [inline Gemfile]: Will you use `byebug` ?

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  Dir[Rails.root.join("spec/support/**/*.rb")].each { |f| require f }
+
+  RSpec.configure do |config|
+  config.fixture_path = "#{::Rails.root}/spec/fixtures"
+  config.use_transactional_fixtures = true
+  config.infer_spec_type_from_file_location!
+  config.filter_rails_from_backtrace!
+  config.include FactoryBot::Syntax::Methods
+  config.include(Shoulda::Matchers::ActiveRecord, type: :model)
```
Comment (tochman) [inline spec/rails_helper.rb]: Please stick to one format. Either leave the parenthesis here and remove above and below or remove them here.

Comment (tochman) [reply spec/rails_helper.rb]: Also, please make sure to format your code before you submit it for review.

File: Gemfile (hunk 1 above)
Comment (johanperjulius1) [reply Gemfile]: No you are right! Thanks! Will change!

File: spec/rails_helper.rb (hunk 2 above)
Comment (johanperjulius1) [reply spec/rails_helper.rb]: > Please stick to one format. Either leave the parenthesis here and remove above and below or remove them here.

Can you specify what you mean?

Course material has it like this:

 spec/rails_helper.rb
...
RSpec.configure do |config|
...
  config.include FactoryBot::Syntax::Methods
  config.include(Shoulda::Matchers::ActiveRecord, type: :model)
end

Thanks!

Comment (tochman) [reply spec/rails_helper.rb]:
```
config.include Shoulda::Matchers::ActiveRecord, type: :model
```

**Pull request merged**